src/VisualStudio/Core/Def/Packaging/PackageInstallerServiceFactory.cs

```diff
@@ -48,9 +48,12 @@ internal partial class PackageInstallerService : AbstractDelayStartedService, IP
         private readonly SVsServiceProvider _serviceProvider;
         private readonly IVsEditorAdaptersFactoryService _editorAdaptersFactoryService;
 
-        // We refer to the package services through proxy types so that we can
-        // delay loading their DLLs until we actually need them.
-        private IPackageServicesProxy _packageServices;
+        private readonly Lazy<IVsPackageInstallerServices> _packageInstallerServices;
+        private readonly Lazy<IVsPackageInstaller2> _packageInstaller;
+        private readonly Lazy<IVsPackageUninstaller> _packageUninstaller;
+        private readonly Lazy<IVsPackageSourceProvider> _packageSourceProvider;
+
+        private ImmutableArray<PackageSource> _packageSources;
```

Comment:
based on hte impl below, why not just use an AsyncLazy here?  It seems clearer and better expressed the intent.  Thoughts?

Reply:
I opted for optimistic concurrency control to avoid unknown interactions between initialization and vs-threading (AsyncLazy is not known to be safe in combination with an application using JTF, so I avoid it unless it provides a benefit not achievable through other means).

Reply:
Gotcha!  Pity about this.  Tagging @jasonmalinowski for visibility.  If we do want to use JTF more in IDE, it would be good if we could get our other core primitives to play nice with it.  AsyncLazy is particularly useful and it's a pity to not be able to use it.

That said, it sounds like AsyncLazy would be wrong here anyways since this is a mutable concurrent value, not a compute-once value.  So i retract the suggestion here anyways.